DataBinding/Realm.DataBinding.PCL/TypeInfoHelper.cs

```diff
@@ -27,8 +27,7 @@ public static class TypeInfoHelper
         [SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1600:ElementsMustBeDocumented")]
         public static TypeInfo GetInfo<T>() where T : RealmObject
         {
-            RealmPCLHelpers.ThrowProxyShouldNeverBeUsed();
-            return null;
+            return typeof(T).GetTypeInfo();
```

Comment:
To elaborate on that - our weaver will weave `IReflectableType` on pcl/net standard projects, because we can't know whether those will be used in an iOS application or a console app. While it's unlikely someone will be casting to `IReflectableType` in a console application, a 3rd party library may be doing it.